src/nextjournal/clerk/analyzer.clj

```diff
@@ -217,6 +218,21 @@
   (-> (reduce add-block-id (assoc analyzed-doc :blocks [] :id->count {} ) blocks)
       (dissoc :id->count)))
 
+(defn ^:private internal-proxy-name?
+  "Returns true if `sym` represents a var name interned by `clojure.core/proxy`."
+  [sym]
+  (str/includes? (name sym) ".proxy$"))
+
+(defn throw-if-dep-is-missing [doc state analyzed]
+  (when-let [missing-dep (and (first (set/difference (into #{}
+                                                           (comp (filter #(and (symbol? %)
+                                                                               (#{(-> state :ns ns-name name)} (namespace %))))
+                                                                 (remove internal-proxy-name?))
+                                                           (:deps analyzed))
+                                                     (-> state :->analysis-info keys set))))]
+    (throw (ex-info (str "The var `#'" missing-dep "` exists at runtime, but Clerk cannot find it in the namespace. Did you remove it?")
+                    (merge {:var-name missing-dep} (select-keys analyzed [:form]) (select-keys doc [:file]))))))
```

Comment:
This is my attempt to integrate the takeaways from https://www.thestrangeloop.com/2022/expert-software-developers-approach-to-error.html into Clerk's development. When I first encountered the error, I didn't even notice it's coming from Clerk (otherwise I would have been able to fix it immediately).

I also experimented with going more into the rationale of this feature and letting folks know why this exists but I feel like this is better explained in longer form elsewhere. Was thinking to include a link at first but it would be even better if Clerk just showed an error where folks can opt into a rich view with more information.